### PR TITLE
13784: Update Personal Information Chapter Layout and questions

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description
As an applicant, I need the completion and review of my Personal Information to be seamless and flow from start to finish so I do not have to review previously answered questions based on questions that follow.

[Originating Issue](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/13784)

Associated vets-website [pull request](https://github.com/department-of-veterans-affairs/vets-api/pull/5003)

## Testing done
Updated unit tests, passing locally.

## Screenshots
Screen Shot 2020-09-30 at 10 39 00 AM

## Acceptance criteria
- [x] The questions in Chapter 5, Page 1 are displayed in the following order:
a. What's the best way for us to contact you?
b. Country
c. Street address
d. Line 2
e. City
f. State
g. Postal code
h. Email address
i. Re-enter email address
j. Home phone number
k. Mobile phone number
l. I would like to receive text messages from VA about my GI Bill benefits

- [x] The following content is displayed after the last question:

Note: Text messages may include status updates or requests to verify school attendance or other GI Bill benefit information. Message and data rates may apply.

- [x] When the user answers the I would like to receive text message from VA about my GI Bill benefits question:
a. The "Mobile phone number" question is marked required and the following message is displayed when triggered "Please enter a mobile phone number".
b. The following alert content is displayed below the question if the user did not enter a phone number prior to checking the box: Since you've opted to receive text message updates, you need to enter your mobile phone number so we can send updates to your device.
c. The alert is no longer displayed once the user enters a valid "Mobile phone number"

- [x] The page matches the layout and structure as specified in Support Artifact #1

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/edu-benefits/10203/helpers.jsx [hunk 1]
```diff
@@ -103,3 +103,8 @@ export function updateProgramDetailsSchema() {
     });
   };
 }
+
+export function phoneNumberFormatted(number) {
```
Comment (zurbergram) [inline src/applications/edu-benefits/10203/helpers.jsx:107]: I don't think this is needed as the phone number is stored in the state as only numbers.
See https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms-system/src/js/widgets/PhoneNumberWidget.jsx#L18

File: src/applications/edu-benefits/10203/pages/personalInformation.js [hunk 2]
```diff
@@ -2,19 +2,27 @@ import fullSchema10203 from 'vets-json-schema/dist/22-10203-schema.json';
...
+
+import { phoneNumberFormatted } from '../helpers';
 
 export const title = contactInformation.title;
 export const path = contactInformation.path;
 export const uiSchema = {
-  'ui:title': 'Your address',
   veteranAddress: {
     ...addressUiSchema,
+    'ui:title': 'Your address',
```
Comment (zurbergram) [inline src/applications/edu-benefits/10203/pages/personalInformation.js:25]:
```suggestion
```

File: src/applications/edu-benefits/10203/pages/personalInformation.js [hunk 3]
```diff
@@ -45,13 +55,46 @@ export const uiSchema = {
...
+        !data?.receiveTexts ||
+        phoneNumberFormatted(data['view:otherContactInfo']?.mobilePhone),
+    },
+  },
+  'view:receiveTextsInfo': {
+    'ui:description': receiveTextsNote,
   },
 };
+
+export const schema = environment.isProduction()
```
Comment (zurbergram) [inline src/applications/edu-benefits/10203/pages/personalInformation.js:75]: There is a production in schema but not in uiSchema

**Review by zurbergram — CHANGES_REQUESTED** [review]
https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms/definitions/address.js#L135 takes a parameter that's allows for changing the address label.

Missing `environement.isProduction()` for `uiSchema` object

File: src/applications/edu-benefits/10203/helpers.jsx (hunk 1 above)
Comment (elder-joseph) [reply src/applications/edu-benefits/10203/helpers.jsx:107]: I added this helper to determine whether or not the alert should be shown given a valid number for 3c. `The alert is no longer displayed once the user enters a valid "Mobile phone number"`. I didnt see anywhere in the state that indicated valid vs invalid for that field.

File: src/applications/edu-benefits/10203/pages/personalInformation.js (hunk 3 above)
Comment (elder-joseph) [reply src/applications/edu-benefits/10203/pages/personalInformation.js:75]: Adding a similar check for UISchema

File: src/applications/edu-benefits/10203/helpers.jsx (hunk 1 above)
Comment (zurbergram) [reply src/applications/edu-benefits/10203/helpers.jsx:107]: https://github.com/department-of-veterans-affairs/vets-json-schema/blob/master/src/common/definitions.js#L159

only validation on phone numbers should be a length check

`fullSchema10203.definitions.phone.minLength` might be the way to access the min length value to check against state data

**Review by zurbergram — APPROVED** [review]

**Review by sonntag-adam — APPROVED** [review]

**Review by rianfowler — APPROVED** [review]

**Pull request merged**